config/v1/types_authentication.go

```diff
@@ -249,8 +253,44 @@ type TokenIssuer struct {
 	// the "ca-bundle.crt" key.
 	// If unset, system trust is used instead.
 	CertificateAuthority ConfigMapNameReference `json:"issuerCertificateAuthority"`
+
+	// discoveryURL, if specified, overrides the URL used to fetch discovery
+	// information instead of using "{url}/.well-known/openid-configuration".
+	// The exact value specified is used, so "/.well-known/openid-configuration"
+	// must be included in discoveryURL if needed.
+	// Must use the https:// scheme.
+	//
+	// +kubebuilder:validation:Pattern=`^https:\/\/[^\s]`
+	// Ensure that discoveryURL is unique within the set of seenDiscoveryURLs
+	// +kubebuilder:validation:XValidation:rule="!seenDiscoveryURLs.has(self.issuer.discoveryURL)",message="discoveryURL must be unique"
+	// +kubebuilder:validation:XValidation:rule="self.issuer.discoveryURL.size() > 0 ? (self.issuer.url.size() == 0 || self.issuer.url.trim('/').notEquals(self.issuer.discoveryURL.trim('/')))" ,message="discoveryURL must be different from URL"
+	// +kubebuilder:validation:XValidation:rule="self.issuer.discoveryURL.size() > 0 ? has(self.StructuredAuthenticationConfiguration) && self.StructuredAuthenticationConfiguration" ,message="discoveryURL is not supported when StructuredAuthenticationConfiguration feature gate is disabled"
+
+	// +optional
+	DiscoveryURL string `json:"discoveryURL,omitempty"`
+
+	// AudienceMatchPolicy controls how the "aud" claim in JWT tokens is validated.
+	// It allows flexible matching of the audience value in tokens.
+	// Possible values: MatchAny, MatchAll.
+	// +kubebuilder:validation:XValidation:rule="self.audienceMatchPolicy.size() > 0",message="At least one audienceMatchPolicy is required"
+	// +kubebuilder:validation:XValidation:rule="self.audienceMatchPolicy.size() <= 1 || (has(self.StructuredAuthenticationConfiguration) && self.StructuredAuthenticationConfiguration)",message="Multiple audienceMatchPolicy values are not supported when StructuredAuthenticationConfiguration feature gate is disabled"
+	// +kubebuilder:validation:XValidation:rule="self.audienceMatchPolicy.all(a, a.size() > 0)",message="AudienceMatchPolicy values cannot be empty"
+	// +kubebuilder:validation:XValidation:rule="self.audienceMatchPolicy.size() == self.audienceMatchPolicy.distinct().size()",message="Duplicate audienceMatchPolicy values are not allowed"
+	// +kubebuilder:validation:XValidation:rule="self.audienceMatchPolicy.size() > 1 ? self.audienceMatchPolicy == 'MatchAny' : true",message="audienceMatchPolicy must be MatchAny for multiple values"
+	// +kubebuilder:validation:XValidation:rule="self.audienceMatchPolicy.size() == 1 ? (self.audienceMatchPolicy == '' || self.audienceMatchPolicy == 'MatchAny') : true",message="audienceMatchPolicy must be empty or MatchAny for a single value"
```

Comment:
These validations don't really make sense to me. AudienceMatchPolicy is a string and not a list so most, if not all of these will not work. 

Looking at https://github.com/kubernetes/kubernetes/blob/83bb5d570580a3f477737fec5c24ba8fc3554264/staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go#L185-L217 I don't think we really need to have any of these validations.

I think simply allowing the following scenarios would be sufficient for the current set of possible options a user has:
- Can always be set to `MatchAny`
- Can always be omitted (i.e set to `""`) , meaning the system chooses a sane default.

I think you can handle this with the standard `// +kubebuilder:validation:Enum` marker

Reply:
I would recommend adding the enum marker on the type alias instead of on this field though